src/refactor_nrepl/find/find_symbol.clj

```diff
@@ -137,7 +137,16 @@
                                (str/join "/" [namespace var-name]))]
     (->> (core/dirs-on-classpath)
          (mapcat (partial core/find-in-dir (some-fn core/clj-file? core/cljc-file?)))
-         (mapcat (partial find-symbol-in-file fully-qualified-name ignore-errors)))))
+         (shuffle) ;; make it less likely that work will be concentrated in one partition (see next line)
```

Comment:
Can't see how that's true.  

The random shuffled seq has no better guarantees of an even split than whatever order is returned when we read from disk.  It's just be different each time.

Reply:
Assume 2 cores and the following workload:

`[:tiny, :tiny, :heavy, :heavy]`

...where `:heavy` takes 2x to process.

Without shuffling, cores would get an uneven distribution 100% of the times, and the overall task performance will be bound by the busiest core.

I would say this is a valid analogy for real-world projects. Those tend to be organised in directories with different purposes and authors, so there's a strong chance that their files will have disparate sizes (and therefore costs).

Reply:
That said one _could_ use a heuristic for attempting a deterministic, even distribution, by querying each file's `.length()` in bytes.

Reply:
> I would say this is a valid analogy for real-world projects. Those tend to be organised in directories with different purposes and authors, so there's a strong chance that their files will have disparate sizes (and therefore costs).

Don't agree with this assumption.  Most organizations have a common code style, e.g. either all files are allowed to grow long, or none are.  If all files are allowed to grow long they'll probably be scattered randomly throughout the project as a byproduct of task complexity.

My objection was to doing extra work and introducing non-determinism (users are left wondering why the op is sometimes fast and other times slow) with no guarantee of a better outcome.

For all we know we read `[:tiny, :heavy, :tiny, :heavy]` from disk and then shuffle it into `[:tiny, :tiny, :heavy, :heavy]`.

Thanks for explaining your thinking.  Since the op was down from 1750ms to 350ms going in either direction is fine with me.

Reply:
Valid objections :) thanks as well!

The last suggestion (`.length`) might be spot on though.

Reply:
To me it makes sense to either remove the `shuffle` or to use the `.length` as a heuristic to re-order the collection.

I agree that it's worth re-ordering the sequence, in the sense that the time it takes to do so will be more than repaid by better parallelization.